Add markdownlinter as a pre-commit hook

arielger opened the pull request:
Add [markdownlint](https://github.com/DavidAnson/markdownlint) as a pre-commit hook using husky and lint-stagged libraries.

Now when you try to commit a markdown file with style errors, you will get something like this:

![image](https://cloud.githubusercontent.com/assets/6318057/26326773/16a47030-3f13-11e7-931d-8e136332e5e4.png)

arielger committed:
Add script to check all markdown files against linter

arielger committed:
Add precommit hook to lint staged markdown files (wip)

arielger committed:
Log linted files and errors

arielger committed:
Add rules to markdownlint

ramasilveyra commented on package.json:
```diff
@@ -5,14 +5,24 @@
   },
   "devDependencies": {
     "commander": "^2.8.1",
+    "filehound": "^1.16.1",
```
Why you need `filehound` ? It's a peerdep?

ramasilveyra commented on package.json:
```diff
@@ -5,14 +5,24 @@
+    "imagemin-optipng": "^4.3.0",
+    "lint-staged": "^3.4.2",
+    "markdownlint": "^0.4.1",
+    "markdownlint-cli": "^0.3.1"
```
Maybe you don't need to list `markdownlint-cli` as devDep.

ramasilveyra commented on check-markdown.js:
```diff
@@ -0,0 +1,54 @@
+const markdownlint = require('markdownlint');
+
+const options = {
+  files: process.argv.slice(2),
```
I think that you will loose some changed files, to properly support running this script from `lint-staged` and from npm script or directly you need to do something like this: https://github.com/auth0/auth0-website/blob/master/tools/scripts/prettier#L36-L54

To debug you can modify `lint-staged` source to see the stdout here https://github.com/okonet/lint-staged/blob/c7283b77c2cf3637e9f06f953b919a556ffde205/src/runScript.js#L19-L20 with:
```js
// ...
.then(result => {
	console.log(result.stdout);
// ...
```

arielger committed:
Add separate config file.

This will enable VS Code to use the markdownlint plugin (https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)

ramasilveyra commented:
With the last commit I believe that you can just use `markdownlint-cli`. 
> CLI argument --config is not mandatory. If it is not provided, markdownlint-cli looks for file .markdownlint.json in current folder, or for file .markdownlintrc in current or all upper folders. The algorithm is described in details on rc package page.

arielger committed:
Replace check-markdown script by markdownlint-cli

- Also remove filehound dependency

arielger committed:
Remove fenced-code-language rule

arielger committed:
Update contributing file with linting info and fix linting errors

arielger committed:
Replace h2 titles by h1 (first-header-h1 rule)

arielger committed:
Replace all h2 main title by h1 (first-header-h1 rule)

mpaktiti committed:
Merge branch 'master' into add-markdownlinter
